test/test_nn.py

```diff
@@ -2864,38 +2864,17 @@ def test_cosine_similarity(self):
         self.assertTrue(gradcheck(lambda x, y: F.cosine_similarity(x, y, dim=-1), (input1, input2)))
 
     def test_grid_sample(self):
-        # test known input on CPU
-        input = Variable(torch.arange(1, 11).view(1, 1, 2, 5))
-        grid = Variable(torch.Tensor(
-            [[-1, -0.5, 0, 0.2, 1],
-             [-1, -0.333, 0, 0.5, 1],
-             [-1, -0.5, 0, 0.3333, 1],
-             [-1, -0.2, 0, 0.2, 1]]).view(1, 2, 5, 2))
-        output = F.grid_sample(input, grid)
-        groundtruth = torch.Tensor(
-            [[2.2500, 6.0000000000, 5.0000, 4.8340, 9.0000],
-             [2.2500, 6.333250045, 5.0000, 5.1000, 8.4000]]).view(1, 1, 2, 5)
-        self.assertEqual(output.data, groundtruth)
+        def test_cpu_against_cuda(N, C, H, W, padding_mode):
```

Comment:
nit:Might be cleaner to just move the cuda tests into their own function i.e 
```python
@unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
def test_grid_sample_cuda(self):
    def test_cpu_against_cuda(N, C, H, W, padding_mode):
            def test_shape(N, C, IH, IW, H, W, padding_mode):
                ....
    for padding_mode in ['zeros', 'border']:
        N = random.randint(1, 8)
        C = random.randint(1, 8)
        H = random.randint(1, 8)
        W = random.randint(1, 8)
        test_cpu_against_cuda(N, C, H, W, padding_mode)
```

But as the version in master bundles them into one test I guess its fine this way too

Reply:
Happy to go ahead with this change if it makes the test cleaner.